src/components/smart_objects/src/array_schema_item.cc

```diff
@@ -89,11 +89,60 @@ Errors::eType CArraySchemaItem::validate(const SmartObject& Object,
   return Errors::OK;
 }
 
-void CArraySchemaItem::applySchema(SmartObject& Object,
-                                   const bool RemoveFakeParameters) {
+Errors::eType CArraySchemaItem::validate(
```

Comment:
This is practically duplicating the body of the other `validate` function with a few minor tweaks, do you think that they can be combined in some way (maybe rewrite the 2-parameter version to use the 3-parameter version)?